### PR TITLE
Emphasize re-register in iOS 13 migration guide

### PR DESCRIPTION
**Pull request by bobiechen-twilio**
- [x] I acknowledge that all my contributions will be made under the project's license.

Updates:
- Add code snippet to show how to register when the app is launched.
- Add a paragraph to emphasize the importance of update and re-register when the app is launched.

### CODE REVIEW

File: iOS-13-Migration-Guide.md [hunk 1]
```diff
@@ -45,7 +45,76 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+	                             deviceToken:self.deviceTokenString
+	                              completion:^(NSError *error) {
+	        ...
+	    }];
+	}
+
+	@end
+	```
+
+This ensures that your app no longer receives “cancel” push notifications. Please note that if the app is updated but never launched for the execution of the above, the mobile client will still receive "cancel" notifications, which could cause the app terminated by iOS if the VoIP push notification is not reported to CallKit as a new incoming call.
```
Comment (idelgado) [inline iOS-13-Migration-Guide.md]: I would still keep this paragraph before the code snippet. Explaining the why should be first.

Comment (idelgado) [inline iOS-13-Migration-Guide.md]: I would create a section under the code snippet that describes what to do to manage the corner case where the app has not been launched yet. Additionally in this section I would actually provide a code snippet with the recommended workaround.

**Review by idelgado — CHANGES_REQUESTED** [review]

File: iOS-13-Migration-Guide.md [hunk 2]
```diff
@@ -45,7 +45,76 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+
+	- (void)setupPushRegistry {
+	    self.voipRegistry = [[PKPushRegistry alloc] initWithQueue:dispatch_get_main_queue()];
+	    self.voipRegistry.delegate = self;
+	    self.voipRegistry.desiredPushTypes = [NSSet setWithObject:PKPushTypeVoIP];
+	}
+
+	#pragma mark - PKPushRegistryDelegate
+	- (void)pushRegistry:(PKPushRegistry *)registry didUpdatePushCredentials:(PKPushCredentials *)credentials forType:(NSString *)type {
+	    self.deviceTokenString = [credentials.token description];
```
Comment (piyushtank) [inline iOS-13-Migration-Guide.md]: I guess we should use the iOS 13 way of getting the device token here?

File: iOS-13-Migration-Guide.md [hunk 3]
```diff
@@ -45,7 +45,76 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+	        ...
+	    }];
+	}
+
+	@end
+	```
+
+This ensures that your app no longer receives “cancel” push notifications. Please note that if the app is updated but never launched for the execution of the above, the mobile client will still receive "cancel" notifications, which could cause the app terminated by iOS if the VoIP push notification is not reported to CallKit as a new incoming call.
+
+A "call" push notification, when passed to `[TwilioVoice handleNotification:delegate:delegateQueue:]`, will return a `TVOCallInvite` object to you synchronously via the `[TVONotificationDelegate callInviteReceived:]` method when `[TwilioVoice handleNotification:delegate:delegateQueue:]` is called. A `TVOCancelledCallInvite` will be raised asynchronously via `[TVONotificationDelegate cancelledCallInviteReceived:error:]` if any of the following events occur:
```
Comment (piyushtank) [inline iOS-13-Migration-Guide.md]: suggestion -
```
A "call" push notification, when passed to `[TwilioVoice handleNotification:delegate:delegateQueue:]`, will return a `TVOCallInvite` object to you synchronously via the `[TVONotificationDelegate callInviteReceived:]` delegate callback. A `TVOCancelledCallInvite` will be raised asynchronously via `[TVONotificationDelegate cancelledCallInviteReceived:error:]` if any of the following events occur:
```

**Review by piyushtank — CHANGES_REQUESTED** [review]

File: iOS-13-Migration-Guide.md [hunk 4]
```diff
@@ -54,26 +54,128 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+            }
+
+            return
+        }
+	}
+	```
+
+	**Objective-C**
+
+	```
```
Comment (idelgado) [inline iOS-13-Migration-Guide.md]: `.objective-c`

File: iOS-13-Migration-Guide.md [hunk 5]
```diff
@@ -54,26 +54,128 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+	}
+
+	@end
+	```
+
+	Please note that if the app is updated but never launched to perform the registration, the mobile client will still receive "cancel" notifications, which could cause the app terminated by iOS if the VoIP push notification is not reported to CallKit as a new incoming call. To workaround and avoid app being terminated on iOS 13, upon receiving a "cancel" notification you can report a dummy incoming call to CallKit with nil UUID:
+
+	**Swift**
+
+	```
```
Comment (idelgado) [inline iOS-13-Migration-Guide.md]: `.swift`

**Review by idelgado — CHANGES_REQUESTED** [review]

**Review by idelgado — COMMENTED** [review]
Once @piyushtank has reviewed this you need to update the 2.1.0 section of the migration guide with similar guidance.

**Review by idelgado — APPROVED** [review]

File: iOS-13-Migration-Guide.md [hunk 6]
```diff
@@ -54,26 +54,128 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+	- (void)setupPushRegistry {
+	    self.voipRegistry = [[PKPushRegistry alloc] initWithQueue:dispatch_get_main_queue()];
+	    self.voipRegistry.delegate = self;
+	    self.voipRegistry.desiredPushTypes = [NSSet setWithObject:PKPushTypeVoIP];
+	}
+
+	#pragma mark - PKPushRegistryDelegate
+	- (void)pushRegistry:(PKPushRegistry *)registry didUpdatePushCredentials:(PKPushCredentials *)credentials forType:(NSString *)type {
+	    const unsigned *tokenBytes = [credentials.token bytes];
+        self.deviceTokenString = [NSString stringWithFormat:@"<%08x %08x %08x %08x %08x %08x %08x %08x>",
```
Comment (piyushtank) [inline iOS-13-Migration-Guide.md:112]: formating.. use space vs tabs I guess

File: iOS-13-Migration-Guide.md [hunk 7]
```diff
@@ -54,26 +54,128 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+
+	```.objective-c
+	- (void)pushRegistry:(PKPushRegistry *)registry
+    didReceiveIncomingPushWithPayload:(PKPushPayload *)payload
+                 forType:(PKPushType)type
+    withCompletionHandler:(void (^)(void))completion {
+        if ([payload.dictionaryPayload[@"twi_message_type"] isEqualToString:@"twilio.voice.cancel"]) {
+        	CXHandle *callHandle = [[CXHandle alloc] initWithType:CXHandleTypeGeneric value:@"alice"];
+
+	        CXCallUpdate *callUpdate = [[CXCallUpdate alloc] init];
```
Comment (piyushtank) [inline iOS-13-Migration-Guide.md]: formatting

File: iOS-13-Migration-Guide.md [hunk 8]
```diff
@@ -54,26 +54,128 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+            callUpdate.supportsDTMF = YES;
+            callUpdate.supportsHolding = YES;
+            callUpdate.supportsGrouping = NO;
+            callUpdate.supportsUngrouping = NO;
+            callUpdate.hasVideo = NO;
+
+            [self.callKitProvider reportNewIncomingCallWithUUID:nil update:callUpdate completion:^(NSError *error) {
+                ...
+            }];
+	        return;
```
Comment (piyushtank) [inline iOS-13-Migration-Guide.md]: formatting..

File: iOS-13-Migration-Guide.md [hunk 9]
```diff
@@ -54,26 +54,128 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
-                                      completion:^(NSError *error) {
-        ...
-    }
-    ```
+	**Swift**
 
+	```.swift
+	func pushRegistry(_ registry: PKPushRegistry, didReceiveIncomingPushWith payload: PKPushPayload, for type: PKPushType, completion: @escaping () -> Void) {
+		if (payload.dictionaryPayload["twi_message_type"] as! String == "twilio.voice.cancel") {
+            let callHandle = CXHandle(type: .generic, value: "alice")
```
Comment (piyushtank) [inline iOS-13-Migration-Guide.md:135]: formatting

File: iOS-13-Migration-Guide.md [hunk 10]
```diff
@@ -54,26 +54,128 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+
+            let callUpdate = CXCallUpdate()
+            callUpdate.remoteHandle = callHandle
+            callUpdate.supportsDTMF = true
+            callUpdate.supportsHolding = true
+            callUpdate.supportsGrouping = false
+            callUpdate.supportsUngrouping = false
+            callUpdate.hasVideo = false
+
+            callKitProvider.reportNewIncomingCall(with: nil, update: callUpdate) { error in
```
Comment (piyushtank) [inline iOS-13-Migration-Guide.md]: This workaround has a risk of app store rejection. We are suggesting that report a call and end immediately where we are not violating Apple's guideline.

Lets use the workaround we are suggesting here - https://github.com/twilio/voice-quickstart-swift/issues/251
cc @idelgado @bchen-twilio

File: iOS-13-Migration-Guide.md [hunk 11]
```diff
@@ -54,26 +54,147 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+	    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+	        ...
+	        self.setupPushRegistry()
+	        ...
+	    }
+
+	    func setupPushRegistry() {
+	        var voipRegistry: PKPushRegistry
+	        voipRegistry.delegate = self
+            voipRegistry.desiredPushTypes = Set([PKPushType.voIP])
```
Comment (piyushtank) [inline iOS-13-Migration-Guide.md:73]: formatting

File: iOS-13-Migration-Guide.md [hunk 12]
```diff
@@ -54,26 +54,147 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+	    }
+
+	    func pushRegistry(_ registry: PKPushRegistry, didUpdate credentials: PKPushCredentials, for type: PKPushType) {
+	        let accessToken = fetchAccessToken()
+            let deviceToken = credentials.token.map { String(format: "%02x", $0) }.joined()
+
+            TwilioVoice.register(withAccessToken: accessToken, deviceToken: deviceToken) { (error) in
```
Comment (piyushtank) [inline iOS-13-Migration-Guide.md:80]: formatting

File: iOS-13-Migration-Guide.md [hunk 13]
```diff
@@ -54,26 +54,147 @@ If your App supports incoming calls, you **MUST** perform the following steps to
...
+            NSUUID *uuid = [NSUUID UUID];
+
+            [self.callKitProvider reportNewIncomingCallWithUUID:uuid update:callUpdate completion:^(NSError *error) {
+                ...
+            }];
+
+            CXEndCallAction *endCallAction = [[CXEndCallAction alloc] initWithCallUUID:uuid];
+            CXTransaction *transaction = [[CXTransaction alloc] initWithAction:endCallAction];
+
+            [self.callKitCallController requestTransaction:transaction completion:^(NSError *error) {
```
Comment (piyushtank) [inline iOS-13-Migration-Guide.md]: dispatch_async(..) here? This should be invoked on the next tick.

**Review by piyushtank — CHANGES_REQUESTED** [review]

**Review by piyushtank — APPROVED** [review]

**Review by idelgado — APPROVED** [review]

**Pull request merged**